classes/OJSSwordDeposit.inc.php

```diff
@@ -0,0 +1,202 @@
+ *
+ * @brief Class providing a SWORD deposit wrapper for OJS submissions
+ */
+
+require_once dirname(__FILE__) . '/../lib/swordappv2/swordappclient.php';
+require_once dirname(__FILE__) . '/../lib/swordappv2/swordappentry.php';
+require_once dirname(__FILE__) . '/../lib/swordappv2/packager_mets_swap.php';
+
+class OJSSwordDeposit {
+	/** @var SWORD deposit METS package */
+	protected $_package = null;
+
+	/** @var Complete path and directory name to use for package creation files */
+	protected $_outPath = null;
+
+	/** @var Journal */
+	protected $_context = null;
+
+	/** @var Section */
+	protected $_section = null;
+
+	/** @var Issue */
+	protected $_issue = null;
+
+	/** @var Article */
+	protected $_article = null;
+
+	/**
+	 * Constructor.
+	 * Create a SWORD deposit object for an OJS article.
+	 * @param $submission Submission
+	 */
+	public function __construct($submission) {
+		$this->_article = $submission;
+
+		// Create a directory for deposit contents
+		$this->_outPath = tempnam('/tmp', 'sword');
+		unlink($this->_outPath);
+		mkdir($this->_outPath);
+		mkdir($this->_outPath . '/files');
+
+		// Create a package
+		$this->_package = new PackagerMetsSwap(
+			$this->_outPath,
+			'files',
+			$this->_outPath,
+			'deposit.zip'
+		);
+
+		$journalDao = DAORegistry::getDAO('JournalDAO');
+		$this->_context = $journalDao->getById($submission->getContextId());
+
+		$sectionDao = DAORegistry::getDAO('SectionDAO');
+		$this->_section = $sectionDao->getById($submission->getSectionId());
+
+		$publishedArticleDao = DAORegistry::getDAO('PublishedArticleDAO');
+		$publishedArticle = $publishedArticleDao->getByArticleId($submission->getId());
+
+		$issueDao = DAORegistry::getDAO('IssueDAO');
+		if ($publishedArticle) {
+			$this->_issue = $issueDao->getById($publishedArticle->getIssueId());
+			$this->_article = $publishedArticle;
+		}
+	}
+
+	/**
+	 * Register the article's metadata with the SWORD deposit.
+	 * @param $request PKPRequest
+	 */
+	public function setMetadata($request) {
+		$this->_package->setCustodian($this->_context->getContactName());
+		$this->_package->setTitle(html_entity_decode($this->_article->getLocalizedTitle(), ENT_QUOTES, 'UTF-8'));
+		$this->_package->setAbstract(html_entity_decode(strip_tags($this->_article->getLocalizedAbstract()), ENT_QUOTES, 'UTF-8'));
+		$this->_package->setType($this->_section->getLocalizedIdentifyType());
+
+		// The article can be published or not. Support either.
+		if (is_a($this->_article, 'PublishedArticle')) {
+			$doi = $this->_article->getStoredPubId('doi');
+			if ($doi !== null) $this->_package->setIdentifier($doi);
+		}
+
+		foreach ($this->_article->getAuthors() as $author) {
+			$creator = $author->getFullName(true);
+			$affiliation = $author->getLocalizedAffiliation();
+			if (!empty($affiliation)) $creator .= "; $affiliation";
+			$this->_package->addCreator($creator);
+		}
+
+// 		TODO check with Nate why cistation style plugin is not enabled by default
```

Comment:
Don't forget this TODO.

Reply:
I have talked to Nate and he doesn't have answers either. So I posted in #pkp-dev.